src/MultiEmailInput.elm

```diff
@@ -41,15 +41,19 @@ type Msg
 type alias State =
     { nextEmail : String
     , id : String
+    , separator : Regex
+    , valid : Regex
     }
 
 
 {-| Initialize the component's state. It needs the component's ID so that it can refocus the textarea when new emails are inserted. By default, we begin with an empty textarea.
 -}
-init : String -> State
-init id =
+init : String -> Regex -> Regex -> State
+init id separator valid =
     { nextEmail = ""
     , id = id
+    , separator = separator
+    , valid = valid
```

Comment:
Are you changing the `id`, `separator` and `valid` based on the component messages? If they are read-only you shouldn't own that data, it's not yours. Let the user provide it to you. The `State` should only be the `nextEmail` really, it's the only data the component controls.

Here's what I had in mind. Your update would look like this:

```elm
update : UpdateConfig a -> Msg -> State -> List a -> ( State, List a, Cmd Msg )
```

So, what is an `UpdateConfig a`?

```elm
type alias UpdateConfig a
    = { id : String
      , toString : a -> String
      , toComparable : a -> comparable
      , separators : [String] -- maybe you could build a valid Regex with this using Regex.escape ?
      }
```

Then, your view would look like:

```elm
view : ViewConfig a -> List (Html.Attribute msg) -> List a -> State -> Html Msg
```

And a `ViewConfig a` would be:

```elm
type alias ViewConfig a
    = { placeholder : String
      , isValid : a -> Bool
      }
```

This way, devs can use your component with ANY `List`, as long as you can perform some basic map operations on its elements!

Then, this repository `elm-multi-email-input` would simply provide functions to build `UpdateConfig` and `ViewConfig` easily. Notice that I would create a new package for the abstraction, called `elm-multi-value-input` or something like that.